### PR TITLE
Fix #27029

### LINKED ISSUE #27029: User without access rights to thirdparties cannot access entry to Knowledge DB
### Bug

User without access rights to thirdparties cannot access entry to Knowledge DB

### Environment Version

18.0

### Environment OS

n/a

### Environment Web server

n/a

### Environment PHP

n/a

### Environment Database

n/a

### Environment URL(s)

knowledgemanagement/knowledgerecord_card.php

### Expected and actual behavior

User should be able to access Knowledge DB also without rights to access thirdparties

### Steps to reproduce the behavior

1. Enable Module Knowledgemanagement
2. Create an entry to Knowledgemanagement
3. Create a user without rights to access thirdparties and read-rights to Knowledgemanagement
4. Access the entry you created in step 2

There will be an error message

### Attached files

_No response_

### PR DESCRIPTION
**Pull request by glu000**
# FIX|Fix #27029

### CODE REVIEW

File: htdocs/core/lib/security.lib.php [hunk 1]
```diff
@@ -1051,7 +1051,7 @@ function checkUserAccessToObject($user, array $featuresarray, $object = 0, $tabl
 				$sql .= " FROM ".MAIN_DB_PREFIX.$dbtablename." as dbt";
 				$sql .= " WHERE dbt.rowid IN (".$db->sanitize($objectid, 1).")";
 				$sql .= " AND dbt.".$dbt_keyfield." = ".((int) $user->socid);
-			} elseif (isModEnabled("societe") && empty($user->rights->societe->client->voir)) {
+			} elseif (isModEnabled("societe") && empty($user->rights->societe->client->voir)&& ($feature != 'knowledgemanagement')) {
```
Comment (eldy) [inline htdocs/core/lib/security.lib.php:1054]: I think a better fix would be to add
'knowledgemanagement' into the array $check line 895. Can you test this ?

Comment (glu000) [reply htdocs/core/lib/security.lib.php:1054]: It works

Comment (glu000) [reply htdocs/core/lib/security.lib.php]: Thanks for this proposal. It works and I updated the PR

Comment (glu000) [reply htdocs/core/lib/security.lib.php]: @eldy May I ask you to merge this PR or tell me what I can do to get it merged

**Pull request merged**